zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/CamundaExporterIT.java

```diff
@@ -382,4 +334,158 @@ private ExporterResourceProvider mockResourceProvider(
 
     return provider;
   }
+
+  @Nested
+  class ExportHandlerTests {
+
+    @TestTemplate
+    void shouldHandleExportedRecords(
+        final ExporterConfiguration config, final SearchClientAdapter clientAdapter)
+        throws IOException {
+      final var provider = new DefaultExporterResourceProvider();
+      provider.init(config);
+
+      final var exporter =
+          createExporter(
+              new HashSet<>(provider.getIndexDescriptors()),
+              new HashSet<>(provider.getIndexTemplateDescriptors()),
+              config);
+      for (final var handler : provider.getExportHandlers()) {
+        exportTest(exporter, handler, clientAdapter);
+      }
+    }
+
+    private <S extends ExporterEntity<S>, T extends RecordValue> Record<T> recordGenerator(
+        final ExportHandler<S, T> handler, final Supplier<Record<T>> createRecord) {
+      for (int i = 0; i < 500; i++) {
+        final var record = createRecord.get();
+        if (handler.handlesRecord(record)) {
+          return record;
+        }
+      }
+      throw new IllegalArgumentException(
+          "Could not generate handled record for " + handler.getClass().getSimpleName());
+    }
+
+    private <S extends ExporterEntity<S>, T extends RecordValue>
+        Supplier<Record<T>> getHandlerRecord(final ExportHandler<S, T> handler) {
```

Comment:
❓ ❌ TBH I'm not fully sure what we are doing here. Or trying to achive here. 

Is this logic because of a missing mapping between handler and record?

Reply:
Basically the `factory` used to generate records is not perfect, all it knows is the field types and it generates a random value for that. For some handlers when they require certain states i the record for example `ListViewFlowNodeFromProcessInstanceHandler` requires in `handlesRecord` that the record is not a process event and its intent is one of 5 intents, maybe over enough random generations the factory randomly generated record would create a record that could be handled but this would result in a flaky test.